src/plugins/score-plugin-scenario/Scenario/Inspector/ObjectTree/SearchReplaceWidget.cpp

```diff
@@ -0,0 +1,327 @@
+      {
+        if(auto state = qobject_cast<const StateModel*>(obj))
+        {
+          listCache.clear();
+          messagesCache.clear();
+
+          auto& root = state->messages().rootNode();
+
+          // First look for addresses containing the looked-up address
+          bool must_add
+              = Process::hasMatchingAddress(root, addresses, listCache, messagesCache);
+
+          // If not found, then look for addresses containing the raw string
+          //          if(!must_add)
+          //            must_add = Process::hasMatchingText(root, stxt, listCache, messagesCache);
+          // FIXME look into state processes?
+
+          // Try to add if the searched text is in the name of the state
+          if(must_add)
+            m_matches.push_back(obj);
+          //          else
+          //            return;
+          //            add_if_contains(*state, stxt, sel);
+        }
+        else if(auto event = qobject_cast<const EventModel*>(obj))
+        {
+          if(State::findAddressInExpression(event->condition(), m_oldAddress))
+          {
+            m_matches.push_back(obj);
+            sel.append(*event);
+          }
+          //            add_if_contains(*event, stxt, sel);
+        }
+        else if(auto ts = qobject_cast<const TimeSyncModel*>(obj))
+        {
+
+          if(State::findAddressInExpression(ts->expression(), m_oldAddress))
+            m_matches.push_back(obj);
+          sel.append(*ts);
+          //            add_if_contains(*ts, stxt, sel);
+        }
+        else if(auto cmt = qobject_cast<const CommentBlockModel*>(obj))
+        { /*
+           if(State::findAddressInExpression(expr,addresses[0].address))
+                    m_matches.push_back(obj);*/
+          //            add_if_contains(*cmt, stxt, sel);
+        }
+        else if(auto interval = qobject_cast<const IntervalModel*>(obj))
+        {
+          for(auto& proc : interval->processes)
+          {
+            selectProcessPortsWithAddress(addresses, proc, sel);
+            for(Process::Inlet* inlet : proc.inlets())
+            {
+              if(sel.contains(inlet))
+              {
+                m_matches.push_back(inlet);
+              }
+            }
+            for(Process::Outlet* outlet : proc.outlets())
+            {
+              if(sel.contains(outlet))
+              {
+                m_matches.push_back(outlet);
+              }
+            }
+            //                        add_if_contains(proc, stxt, sel);
+            sel.append(proc);
+          }
+
+          //          add_if_contains(*interval, stxt, sel);
+        }
+      }
+    }
+    score::SelectionDispatcher d{doc->context().selectionStack};
+    d.select(sel);
+  }
+}
+
+void SearchReplaceWidget::replace()
+{
+  auto& ctx = m_ctx.documents.currentDocument()->context();
+  MacroCommandDispatcher<Command::ReplaceAddresses> disp{ctx.commandStack};
+  if(!m_newAddress.isSet())
+  {
+    return;
+  }
+  else
```

Comment:
in this case (`if(xx) return;`) don't put a else, just put it after, this way there is no unneeded indentation level. 
this even has a name: return-early pattern https://medium.com/swlh/return-early-pattern-3d18a41bba8  (and it's a good one, i try to use it as much as possible)